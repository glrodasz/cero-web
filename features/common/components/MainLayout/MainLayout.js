import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'

const MainLayout = ({ menu, content, isPlayground }) => {
  return (
    <>
      <main
        className={classNames('main-layout', { 'is-playground': isPlayground })}
      >
        <section className="menu">{menu}</section>
        <section className="content">{content}</section>
      </main>
      <style jsx>{`
        main > section {
          width: 100%;
        }

        .is-playground > section {
          border: var(--border-width-thick) dashed var(--color-primary);
        }

        .main-layout {
          display: flex;
          flex-direction: column-reverse;
          width: 100%;
          height: 100vh;
          justify-content: space-between;
        }

        .menu {
          background: var(--background-color-primary-highlight);
        }

        .content {
          background: var(--background-color-primary);
        }

        @media (min-width: 992px) {
          .main-layout {
            flex-direction: row;
          }

          .menu {
            max-width: 260px;
            height: 100%;
          }
        }
      `}</style>
    </>
  )
}

MainLayout.propTypes = {
  menu: PropTypes.node,
  content: PropTypes.node,
  isPlayground: PropTypes.bool,
}

export default MainLayout
