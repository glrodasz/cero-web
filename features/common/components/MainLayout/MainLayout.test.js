import MainLayout from './MainLayout'
import { render } from '@testing-library/react'

describe('[ features / common / components / MainLayout ]', () => {
  describe('when `MainLayout` is mounted', () => {
    it('should render', () => {
      // Arrange
      const props = {
        menu: 'menu-component',
        content: 'content-component',
        isPlayground: true,
      }

      // Act
      const { asFragment } = render(<MainLayout {...props} />)

      // Assert
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
