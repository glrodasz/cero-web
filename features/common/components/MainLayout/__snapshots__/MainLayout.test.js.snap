// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ features / common / components / MainLayout ] when \`MainLayout\` is mounted should render 1`] = `
<DocumentFragment>
  <main
    class="jsx-1470150938 main-layout is-playground"
  >
    <section
      class="jsx-1470150938 menu"
    >
      menu-component
    </section>
    <section
      class="jsx-1470150938 content"
    >
      content-component
    </section>
  </main>
</DocumentFragment>
`;
